cluster_mapping_rules.go

```diff
@@ -68,55 +63,50 @@ func NewClusterConfigChange(c Cluster) ClusterConfigChange {
 	}
 }
 
-// A Mapping is a set of rules mapping shards onto a storage cluster.
-type Mapping interface {
+// A ClusterMappingRuleProvider provides cluster mapping rules
+type ClusterMappingRuleProvider interface {
 	// RulesForShard returns the mapping rules that currently apply to the given shard, plus
 	// a channel that can be used to receive new mapping rules
-	RulesForShard(n int) ([]MappingRule, <-chan []MappingRule, error)
-
-	// Clusters returns all of the currently defined clusters, plus a channel
-	// that can be used to receive cluster changes (new clusters, changes to
-	// cluster configurations)
-	Clusters() ([]Cluster, <-chan ClusterConfigChange, error)
+	RulesForShard(n int) ([]ClusterMappingRule, <-chan []ClusterMappingRule, error)
```

Comment:
Another thing is, should we perhaps make this just a `Rules()` and sends the shard along with each update.  That way it can be called just once instead of once per shard and rules come over the channel.